addons/base_import_module/models/ir_module.py

```diff
@@ -25,6 +25,10 @@ class IrModule(models.Model):
 
     imported = fields.Boolean(string="Imported Module")
 
+    @classmethod
+    def _get_modules_to_load_query(cls):
+        return super()._get_modules_to_load_query() + " AND imported IS NOT TRUE"
+
```

Comment:
I would change the extension of the query to so that you can only extend the where clause, and hardcode the select and from. IMHO this is more idiot proof

Reply:
Use a ORM domain instead?